### ISSUE TITLE
Telephone - required field isn't correct set in template (Fix included)

### ISSUE THREAD

**Issue by Triloworld** [magento/magento2#12891]
### Preconditions
1. Magento 2.2.0
2. Working site

### Steps to reproduce
1. Open in editor:
vendor/magento/module-customer/view/frontend/templates/widget/telephone.phtml
2. Input field get required class from fax!
3. Not possible to override by just setting flag for field. Need to be fixed in template first.

### Expected result
1. Change field "Telephone" to required field and it work without editing template.

### Actual result
1. Is validated in backend. Not in frontend because of bad validation class in template.

Just check my method to fix it and someone can say it is ok and why not.

TEMPLATE: /vendor/magento/module-customer/view/frontend/templates/widget/telephone.phtml
BAD - IN CORE:
```
<div class="field telephone <?= $block->isRequired() ? 'required' : '' ?>">
    <label for="telephone" class="label">
        <span>
            <?= $block->escapeHtml(__('Phone Number')) ?>
        </span>
    </label>
    <div class="control">
        <?php
            $_validationClass = $block->escapeHtmlAttr(
                $this->helper('Magento\Customer\Helper\Address')
                     ->getAttributeValidationClass('fax')
            );
        ?>
        <input type="text"
               name="telephone"
               id="telephone"
               value="<?= $block->escapeHtmlAttr($block->getTelephone()) ?>"
               title="<?= $block->escapeHtmlAttr(__('Phone Number')) ?>"
               class="input-text <?= $_validationClass ?: '' ?>"
        >
    </div>
</div>
```

GOOD - MY FIX BY OVERRIDE TEMPLATE: 
```
<div class="field telephone <?= $block->isRequired() ? 'required' : '' ?>">
    <label for="telephone" class="label">
        <span>
            <?= $block->escapeHtml(__('Phone Number')) ?>
        </span>
    </label>
    <div class="control">
        <?php
            $_validationClass = $block->escapeHtmlAttr(
                $this->helper('Magento\Customer\Helper\Address')
                     ->getAttributeValidationClass('telephone')
            );
        ?>
        <input type="text"
               name="telephone"
               id="telephone"
               value="<?= $block->escapeHtmlAttr($block->getTelephone()) ?>"
               title="<?= $block->escapeHtmlAttr(__('Phone Number')) ?>"
               class="input-text <?= $_validationClass ?: '' ?>"
        >
    </div>
</div>
```

**Comment by PieterCappelle** [comment]
Looks good. Please create PR so Magento can merge.

**Comment by magento-engcom-team** [comment]
@Triloworld, thank you for your report.
The issue is already fixed in 2.2.2
